packages/widget-chat/src/actions/flags.js

```diff
@@ -0,0 +1,98 @@
+export const ADD_FLAG = `ADD_FLAG`;
+function actionAddFlag(activity) {
+  return {
+    type: ADD_FLAG,
+    activity
+  };
+}
+
+export const BEGIN_RECEIVE_FLAGS = `BEGIN_RECEIVE_FLAGS`;
+function actionBeginReceiveFlags() {
+  return {
+    type: BEGIN_RECEIVE_FLAGS
+  };
+}
+
+export const RECEIVE_FLAGS = `RECEIVE_FLAGS`;
+function actionReceiveFlags(flags) {
+  return {
+    type: RECEIVE_FLAGS,
+    flags
+  };
+}
+
+export const REMOVE_FLAG = `REMOVE_FLAG`;
+function actionRemoveFlag(flag) {
+  return {
+    type: REMOVE_FLAG,
+    flag
+  };
+}
+
+export const REMOVE_FLAG_FAIL = `REMOVE_FLAG_FAIL`;
```

Comment:
Similar to comment above. The redux docs highlight 2 ways of doing this: http://redux.js.org/docs/advanced/AsyncActions.html.
Just personal preference, but `REMOVE_FLAG_FAIL` just doesn't sound like an action to me, but this is just me nitpicking semantics at this point.